### PR TITLE
Fix #1470: Port SearchBar from Gatsby -> NextJS

### LINKED ISSUE #1470: [next] Port SearchBar from Gatsby to NextJS
**What would you like to be added**:
This is an issue to port SearchBar from Gatsby to Next.js

**Why would you like this to be added**:
Part of [#1316](https://github.com/Seneca-CDOT/telescope/issues/1316), Next.js migration project.

### PR DESCRIPTION
**Pull request by MSTEWARDSON**
## Issue This PR Addresses

Fixes #1470 

## Type of Change

- [ ] **Bugfix**: Change which fixes an issue
- [x] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [ ] **UI**: Change which improves UI

## Description

Part of the Gatsby port to NextJS project. This PR handles the porting of the SearchBar component from Gatsby + JS to NextJS + TS. In general here is what has changed:

- Removed unnecessary files
                  - src/frontend/.cache
                  - src/frontend/public
                  - src/frontend/node_modules
- Added "components" folder to contain the new SearchBar.tsx
- Updated eslintrc.js to have a typescript plugin
- Converted SearchBar component to NextJS
- Installed missing packages in the next frontend

## Checklist

- [ ] **Quality**: This PR builds and passes our npm test and works locally
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [ ] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

### CODE REVIEW

**Review by c3ho — CHANGES_REQUESTED** [review]

File: src/frontend/next/src/components/SearchBar/SearchBar.tsx [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+  selectItem: {
+    fontSize: '1.4rem',
+    textTransform: 'capitalize',
+    color: theme.palette.primary.main,
+  },
+}));
+
+interface CustomizedInputBaseProps {
+  text: string | null;
+  onTextChange(): any;
```
[the remark this answers is not in the document]
Comment (MSTEWARDSON) [reply src/frontend/next/src/components/SearchBar/SearchBar.tsx:82]: Yep just checked, it should be void in this case. Thanks.

File: package.json [hunk 2]
```diff
@@ -98,6 +98,7 @@
     "ioredis": "4.17.3",
     "ioredis-mock": "5.1.0",
     "jsdom": "16.4.0",
+    "next": "10.0.3",
```
[the remark this answers is not in the document]
Comment (MSTEWARDSON) [reply package.json:101]: Yeah I think that's what happened here. I will remove this.

File: src/frontend/next/package.json [hunk 3]
```diff
@@ -8,7 +8,10 @@
     "start": "next start"
   },
   "dependencies": {
+    "@material-ui/core": "^4.11.2",
+    "@material-ui/icons": "^4.11.2",
     "next": "^10.0.2",
+    "prop-types": "^15.7.2",
```
Comment (c3ho) [inline src/frontend/next/package.json:14]: Forgot about this, are we still using prop-types? I don't see it anywhere in the code below

Comment (MSTEWARDSON) [reply src/frontend/next/package.json:14]: I added that back into the next frontend early on, I don't believe its needed anywhere now so ill remove it.

**Comment by tonyvugithub** [comment]
@MSTEWARDSON : Hi there, we are in the process of reviewing these migrated to Next PRs. Just wonder if you are still interested to work on this as we would like you to make some changes to your PR. Please let us know in case you decided to drop this then we can assign the issue to others. Thank you.

**Comment by MSTEWARDSON** [comment]
@tonyvugithub Hey, atm I'm quite busy. If someone else wants to take this over this that's fine by me.

**Comment by tonyvugithub** [comment]
@MSTEWARDSON : got it. Thanks

**Comment by PedroFonsecaDEV** [comment]
Working on it.

**Comment by PedroFonsecaDEV** [comment]
PR Fixes #1470: Port SearchBar to NEXT: https://github.com/Seneca-CDOT/telescope/pull/1581

**Pull request closed without merging**